### PR TITLE
fix: event Duration displays incorrectly in event name

### LINKED ISSUE #16887: Event Duration displays incorrectly in event name
Found a bug? Please fill out the sections below. 👍

### Issue Summary

I used the advance option to name an event automatically with the following:

Video Call {Event duration}: {Organiser} and {Scheduler}

For some reason when people schedule a 30 min video call, it displays 60 min on the calendar invite

### Steps to Reproduce

1. Go to my cal page for a video event: https://cal.com/d/wimyZ3D94LjGG7Wuzn8CGh/vid
2. Choose 30 min for duration
3. Schedule an event and see the event name says 60 min

### Actual Results

The event name seems to always say 60 min

### Expected Results

The event name should say whatever the time chosen is as the event duration

### Technical details

This is happening when people schedule with me and has happened from 2 different people - I don't know their browser set up

### Evidence

See screenshot - both are 30 min meetings and both event names say 60

### PR DESCRIPTION
**Pull request by divyanshchahar**
## What does this PR do?

- Fixes #16887
 
[Loom video of the fix](https://www.loom.com/share/0d01be1f663f45e3920645e2a503d41c?sid=c5c0ac1b-7dec-43c6-b9be-b6aa49b63389)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Steps to test the fix (see loom video)
1. Create an event with the following attributes
1.1. event has multiple duration
1.2. booker is allowed to select duration
1.3 under advanced tab of the event the `with event name` field is setup something like this `Event Name {Event duration}: {Organiser} and {Scheduler}`
2. Book this event with any thing but the default duration
3. Check `baseaddress/booking/{bookingid}` 
4. The name of the event **should include the selected duration**
5. The email **should include the selected duration**

___

- Are there environment variables that should be set ?  **no**
- What are the minimal test data to have? A regular user and an event as described above
- What is expected (happy path) to have (input and output)? ** name of the event is in line with selected non default duration everywhere**
- Any other important info that could help to test that PR ? **no**

## What caused the bug 

`getEventName` function in `./packages/core/event.ts` was computing event name based on the default length of the event

## How was the issue fixed ?
- `getEventName` was updated to accept two additional optional parameters (start time and end time) and if these parameters were passed then the duration in the event name is computed from these two additional parameters instead of using the default value

- function calls to the above function were updated to pass the two new arguments

## Environment

production

### CODE REVIEW

File: packages/core/event.ts [hunk 1]
```diff
@@ -53,6 +59,11 @@ export function getEventName(eventNameObj: EventNameObjectType, forAttendeeView
     eventName = eventName.replace("{LOCATION}", locationString);
   }
 
+  const duration =
+    startTime && endTime
+      ? `${dayjs(endTime).diff(dayjs(startTime), "minute")} mins`
+      : `${String(eventNameObj.eventDuration)} mins`;
+
```
Comment (anikdhabal) [inline packages/core/event.ts:66]: We are not fixing the actual bug; instead, we are adding a solution on top of it just to show the event duration. I don't think this is a good way to fix it. It used to work before. Could you check that more deeply?

**Review by anikdhabal — CHANGES_REQUESTED** [review]
@divyanshchahar thanks for your work. Left a comment

**Comment by divyanshchahar** [comment]
@anikdhabal can you please provide some feedback on my latest comments

**Comment by anikdhabal** [comment]
@divyanshchahar thanks for your work. but I think it was fixed.  Let me know if you can reproduce it

**Pull request closed without merging**